pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -543,4 +556,41 @@ private boolean isValid(Schema schema, IndexingConfig indexingConfig) {
     }
     return isValid;
   }
+
+  private int getSegmentPartitionId(String segmentName, String tableName) {
```

Comment:
this logic is repeated in `RealtimeSegmentAssignment.java`?

Reply:
Refactor to a common util class.